libs/perun/components/src/lib/selection-item-search-select/selection-item-search-select.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { AttributeDefinition } from '@perun-web-apps/perun/openapi';
+
+export class SelectionItem {
+  value: string;
+  displayName: string;
+
+  constructor(displayName: string, value: string) {
+    this.value = value;
+    this.displayName = displayName;
+  }
+}
+
+export enum ItemType {
+  SOURCE,
+  DESTINATION,
+  FEDERATION,
+}
+
+@Component({
+  selector: 'perun-web-apps-selection-item-search-select',
+  templateUrl: './selection-item-search-select.component.html',
+  styleUrls: ['./selection-item-search-select.component.scss']
+})
+export class SelectionItemSearchSelectComponent implements OnInit{
+
+  constructor(private translateService: TranslateService) { }
+
+  @Input()
+  attributes: AttributeDefinition[] = [];
+
+  @Input()
+  selectedAttribute: string;
+
+  @Input()
+  type: ItemType;
+
+  @Input()
+  asGroup = false;
+
+  @Output()
+  itemSelected = new EventEmitter<SelectionItem>();
+
+  items: SelectionItem[] = [];
+  item: SelectionItem;
+
+  ngOnInit(): void {
+    if (this.type === ItemType.FEDERATION){
+      this.getFederationAttributes();
+      this.getFederationAttribute();
+    } else {
+      this.createSelectionItems();
+    }
+    this.sortAttributes(this.items);
+  }
+
+  nameFunction = (item: SelectionItem) => item.displayName;
+  shortNameFunction = (item: SelectionItem) => null;
+  searchFunction = (item: SelectionItem) => item.displayName;
+
+  createSelectionItems() {
+    this.translateService.get('DIALOGS.APPLICATION_FORM_EDIT_ITEM.NO_SELECTED_ITEM').subscribe( noItem => {
+      const emptyItem: SelectionItem = new SelectionItem(noItem, '');
+      this.items.push(emptyItem);
+      this.item = emptyItem;
+    });
+
+    for (const attribute of this.attributes) {
+      const item: SelectionItem = new SelectionItem(attribute.friendlyName + ' (' + attribute.entity + ' / ' + SelectionItemSearchSelectComponent.getDefinition(attribute) + ')',
+        attribute.namespace + ':' + attribute.friendlyName);
+      if(item.value === this.selectedAttribute){
+        this.item = item;
+      }
+
+      if (attribute.entity.toLowerCase() === 'user' || attribute.entity.toLowerCase() === 'member') {
+        // add only member and user attributes
+        this.items.push(item);
+      } else if (attribute.entity.toLowerCase() === 'vo' && this.type) {
```

Comment:
```suggestion
      } else if (attribute.entity.toLowerCase() === 'vo' && this.type === "SOURCE") {
```
I think this is what you wanted to type